modules/market/src/main/java/com/opengamma/strata/market/surface/DefaultSurfaceMetadataBuilder.java

```diff
@@ -0,0 +1,277 @@
+   * removing the key if the day count is null.
+   * 
+   * @param dayCount  the day count, may be null
+   * @return this, for chaining
+   */
+  public DefaultSurfaceMetadataBuilder dayCount(DayCount dayCount) {
+    return addInfo(SurfaceInfoType.DAY_COUNT, dayCount);
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Adds a single piece of additional information.
+   * <p>
+   * This is stored in the additional information map using {@code Map.put} semantics,
+   * removing the key if the instance is null.
+   * 
+   * @param <T>  the type of the info
+   * @param type  the type to store under
+   * @param instance  the instance to store, may be null
+   * @return this, for chaining
+   */
+  public <T> DefaultSurfaceMetadataBuilder addInfo(SurfaceInfoType<T> type, T instance) {
+    ArgChecker.notNull(type, "type");
+    if (instance != null) {
+      this.info.put(type, instance);
+    } else {
+      this.info.remove(type);
+    }
+    return this;
+  }
+
+  /**
+   * Adds additional information.
+   * <p>
+   * This is stored in the additional information map using {@code Map.putAll} semantics
+   * 
+   * @param info  the information to add
+   * @return this, for chaining
+   */
+  public DefaultSurfaceMetadataBuilder addInfo(Map<SurfaceInfoType<?>, Object> info) {
+    ArgChecker.notNull(info, "infoMap");
+    this.info.putAll(info);
+    return this;
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Adds a single piece of parameter metadata.
+   * <p>
+   * This is stored in the parameter metadata list.
+   * 
+   * @param parameterMetadata  the parameter metadata
+   * @return this, for chaining
+   */
+  public DefaultSurfaceMetadataBuilder addParameterMetadata(SurfaceParameterMetadata parameterMetadata) {
+    if (this.parameterMetadata == null) {
+      this.parameterMetadata = new ArrayList<>();
+    }
+    this.parameterMetadata.add(parameterMetadata);
+    return this;
+  }
+
+  /**
+   * Sets the metadata about the parameters.
+   * <p>
+   * If present, the parameter metadata will match the number of parameters on the surface.
+   * <p>
+   * This will replace all existing data in the metadata list.
+   * 
+   * @param parameterMetadata  the parameter metadata, may be null
+   * @return this, for chaining
+   */
+  public DefaultSurfaceMetadataBuilder parameterMetadata(List<? extends SurfaceParameterMetadata> parameterMetadata) {
+    if (parameterMetadata == null) {
+      this.parameterMetadata = null;
+    } else {
+      if (this.parameterMetadata == null) {
+        this.parameterMetadata = new ArrayList<>();
+      }
+      this.parameterMetadata.clear();
+      this.parameterMetadata.addAll(parameterMetadata);
+    }
+    return this;
+  }
+
+  /**
+   * Sets the {@code parameterMetadata} property in the builder from an array of objects.
```

Comment:
"from an array of objects" is true but that's just an implementation detail of varags. It seems an odd way to describe what's happening here.